### PR TITLE
Hardening (day 5)

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: src/main/java/org/sonarlint/intellij/ui/RecentProjectsPanel.kt [hunk 1]
```diff
@@ -183,7 +183,7 @@ open class SonarLintRecentProjectPanel(private val onProjectSelected: (Project)
     }
 
     init {
-        var recentProjectActions = RecentProjectsManager.getInstance().getRecentProjectsActions(false, false).toList()
+        var recentProjectActions = RecentProjectListActionProvider.getInstance().getActions(false, false).toList()
```
Comment (jblievremont) [inline src/main/java/org/sonarlint/intellij/ui/RecentProjectsPanel.kt:186]: Looks like this class was only [introduced in 2020.1](https://github.com/JetBrains/intellij-community/commit/8c4e82f42915f98af74b72fba32ceb8f0c2e012c) 😕

**Review by jblievremont — CHANGES_REQUESTED** [review]

File: src/main/java/org/sonarlint/intellij/ui/RecentProjectsPanel.kt (hunk 1 above)
Comment (henryju) [reply src/main/java/org/sonarlint/intellij/ui/RecentProjectsPanel.kt:186]: Indeed, the QA is red on 2019.3 axis. I will revert this change.

File: src/main/java/org/sonarlint/intellij/trigger/EditorChangeTrigger.java [hunk 2]
```diff
@@ -68,7 +68,7 @@ public void started(AnalysisRequest analysisRequest) {
         }
       });
     watcher.start();
-    EditorFactory.getInstance().getEventMulticaster().addDocumentListener(this);
+    EditorFactory.getInstance().getEventMulticaster().addDocumentListener(this, this);
```
[the remark this answers is not in the document]
Comment (henryju) [reply src/main/java/org/sonarlint/intellij/trigger/EditorChangeTrigger.java:71]: This is a parentDisposable. When the parent object (=this) is disposed, then the listener (=this) will be automatically removed from the list of document listeners. I think this code is only sensitive for dynamic unloading, so we will have to retest it anyway when reworking on the topic.

File: src/main/java/org/sonarlint/intellij/server/SonarLintHttpServer.kt [hunk 3]
```diff
@@ -119,25 +121,29 @@ class ServerHandler : SimpleChannelInboundHandler<Any?>() {
...
-            response = RequestProcessor().processRequest(Request(msg.uri(), msg.method()))
+            response = RequestProcessor().processRequest(Request(msg.uri(), msg.method(), isTrustedOrigin(origin)))
         }
         if (msg is LastHttpContent) {
             ctx.writeAndFlush(createResponse(response, origin))
         }
     }
 
+    private fun isTrustedOrigin(origin: String?): Boolean {
+        return origin != null && (SonarLintUtils.isSonarCloudAlias(origin) || !Settings.getGlobalSettings().getConnectionsTo(origin).isEmpty())
```
Comment (jblievremont) [inline src/main/java/org/sonarlint/intellij/server/SonarLintHttpServer.kt:132]: Spotted an issue while testing: the condition in `getConnectionsTo(String)` is not the right one for this need.

If the server is hosted under a directory - e.g http://some.host/dir - then the `Origin` will be http://some.host (without `/dir`). According to the `Origin` header specification, the correct check in our case would be to find a server connection whose `hostUrl` starts with the passed `Origin`.

**Review by jblievremont — CHANGES_REQUESTED** [review]
Manually tested:
* Secrets in CLion and Rider 👍🏻  
* Check on trusted servers 👎🏻 - see my comment

**Review by jblievremont — APPROVED** [review]
LGTM! 👍🏻

**Pull request merged**